MediaBrowser.MediaEncoding/Subtitles/SubtitleEditParser.cs

```diff
@@ -57,5 +85,94 @@ public SubtitleTrackInfo Parse(Stream stream, CancellationToken cancellationToke
             trackInfo.TrackEvents = trackEvents;
             return trackInfo;
         }
+
+        /// <inheritdoc />
+        public bool SupportsFileExtension(string fileExtension)
+            => _subtitleFormats.ContainsKey(fileExtension);
+
+        private IEnumerable<SubtitleFormat> GetSubtitleFormats()
+        {
+            var subtitleFormats = new List<SubtitleFormat>();
+            var assembly = Assembly.GetAssembly(typeof(SubtitleFormat));
+            if (assembly == null)
+            {
+                _logger.LogError("Missing assembly containing {SubtitleFormatName}", nameof(SubtitleFormat));
+                return GetFallbackSubtitleFormats();
+            }
+
+            foreach (var type in assembly.GetTypes())
+            {
+                if (!type.IsSubclassOf(typeof(SubtitleFormat)))
```

Comment:
Should we check to make sure it isn't abstract?
```suggestion
                if (!type.IsSubclassOf(typeof(SubtitleFormat)) || type.IsAbstract)
```